linux57-tkg/install.sh

```diff
@@ -44,17 +47,46 @@ _misc_adds="false" # We currently don't want this enabled on non-Arch
 
 if [ "$1" = "install" ] || [ "$1" = "config" ]; then
 
-  source linux*-tkg-config/prepare
+  if [ -z $_distro ]; then
+    
+    echo "Which linux distribution are you running ?"
+    echo "if it's not on the list, chose the closest one to it: Fedora/Suse for RPM, Ubuntu/Debian for DEB"
+    echo "   1) Debian"
+    echo "   2) Fedora"
+    echo "   3) Suse"
+    echo "   4) Ubuntu"
+    read -p "[1-4]: " _distro_index
+
+    if [ "$_distro_index" = "1" ]; then
+      _distro="Debian"
+    elif [ "$_distro_index" = "2" ]; then
+      _distro="Fedora"
+    elif [ "$_distro_index" = "3" ]; then
+      _distro="Suse"
+    elif [ "$_distro_index" = "4" ]; then
+      _distro="Ubuntu"
+    else
+      echo "Wrong index. aborting..."
+      exit 0
+    fi
 
-  if [ $1 = "install" ] && [ "$_distro" != "Ubuntu" ]; then
-    msg2 "Variable \"_distro\" in \"customization.cfg\" hasn't been set to \"Ubuntu\""
-    msg2 "This script can only install custom kernels for Ubuntu and Debian derivatives. Exiting..."
+  fi
+
+  if [[ $1 = "install" && "$_distro" != "Ubuntu" && "$_distro" != "Debian" &&  "$_distro" != "Fedora" && "$_distro" != "Suse" ]]; then 
+    msg2 "Variable \"_distro\" in \"customization.cfg\" hasn't been set to \"Ubuntu\", \"Debian\",  \"Fedora\" or \"Suse\""
+    msg2 "This script can only install custom kernels for RPM and DEB based distros, though only those keywords are permitted. Exiting..."
     exit 0
   fi
 
-  if [ "$_distro" = "Ubuntu" ]; then
+  if [ "$_distro" = "Ubuntu" ] || [ "$_distro" = "Debian" ]; then
     msg2 "Installing dependencies"
-    sudo apt install git build-essential kernel-package fakeroot libncurses5-dev libssl-dev ccache bison flex
+    sudo apt install git build-essential kernel-package fakeroot libncurses5-dev libssl-dev ccache bison flex qtbase5-dev -y
+  elif [ "$_distro" = "Fedora" ]; then
+    msg2 "Installing dependencies"
+    sudo dnf install fedpkg fedora-packager rpmdevtools ncurses-devel pesign grubby qt5-devel libXi-devel gcc-c++ git ccache flex bison elfutils-libelf-devel openssl-devel dwarves rpm-build -y
+  elif [ "$_distro" = "Suse" ]; then
+    msg2 "Installing dependencies"
+    sudo zypper install -y rpmdevtools ncurses-devel pesign libXi-devel gcc-c++ git ccache flex bison elfutils libelf-devel openssl-devel dwarves make patch bc rpm-build libqt5-qtbase-common-devel libqt5-qtbase-devel lz4
   else
     msg2 "Dependencies are unknown for the target linux distribution."
```

Comment:
Since you've added the check for `$_distro` being one of the valid choices above, this code path shouldn't be reachable, and can probably be removed.

Reply:
oh right, that reminds me: I completely forgot about the other distros, they should be able to do `./install.sh config` without having to worry about giving out their distro type. I will fix that